lib/failure.sh

```diff
@@ -361,6 +361,77 @@ fail_invalid_semver() {
   fi
 }
 
+# Yarn 2 failures
+
+fail_missing_yarnrc_yml() {
+  local build_dir="$1"
+
+  if [[ ! -f "$build_dir/.yarnrc.yml" ]]; then
+    mcount "failures.missing-yarnrc-yml"
+    meta_set "failure" "missing-yarnrc-yml"
+    header "Build failed"
+    warn "The yarnrc.yml file could not found
```

Comment:
Nitpick: This is an error, not a warning. Probably suboptimal naming of the logging functions and not related to this PR?

Reply:
Agreed - it makes sense to do some work that lists this as an error with a failure rather than a warn given that one will fail the build and one won't.